src/components/profile/UserProfilePage/index.tsx

```diff
@@ -0,0 +1,192 @@
+import EditIcon from '@/public/assets/icons/edit.svg';
+import FacebookIcon from '@/public/assets/icons/facebook-icon.svg';
+import GithubIcon from '@/public/assets/icons/github-icon.svg';
+import InstagramIcon from '@/public/assets/icons/instagram.svg';
+import LeaderboardIcon from '@/public/assets/icons/leaderboard-icon.svg';
+import LinkedinIcon from '@/public/assets/icons/linkedin-icon.svg';
+import MajorIcon from '@/public/assets/icons/major-icon.svg';
+import ProfileIcon from '@/public/assets/icons/profile-icon.svg';
+import { Tooltip } from '@mui/material';
+import Link from 'next/link';
+import { useEffect, useState } from 'react';
+import { AiOutlineLink } from 'react-icons/ai';
+import { IoMail } from 'react-icons/io5';
+import styles from './style.module.scss';
+
+export interface UserProfilePageProps {
+  handleUser: PublicProfile;
+  isSignedInUser: boolean;
+  signedInAttendances: PublicAttendance[];
+  attendances?: PublicAttendance[];
+}
+
+const socialMediaIcons = {
+  LINKEDIN: LinkedinIcon,
+  GITHUB: GithubIcon,
+  DEVPOST: DevpostIcon,
+  PORTFOLIO: AiOutlineLink,
+  FACEBOOK: FacebookIcon,
+  INSTAGRAM: InstagramIcon,
+  EMAIL: IoMail,
+};
+
+export const UserProfilePage = ({
+  handleUser,
+  attendances,
+  signedInAttendances,
+  isSignedInUser,
+}: UserProfilePageProps) => {
+  // animate the progress bar
+  const [progress, setProgress] = useState<Number>(0);
+  useEffect(() => setProgress(handleUser.points % 100), [handleUser.points]);
+
+  return (
+    <div className={styles.profilePage}>
+      <div className={styles.cardWrapper}>
+        <div className={styles.banner} />
+        <div className={styles.profileCard}>
+          <div className={styles.profilePic}>
+            <GifSafeImage
+              src={getProfilePicture(handleUser)}
+              alt="Profile Picture"
+              height={112} // 7 rem
+              width={112} // 7 rem
+              priority
+              className={styles.profilePic}
+            />
+          </div>
+          <div className={styles.cardName}>
+            <Typography
+              variant="h1/bold"
+              component="h1"
+            >{`${handleUser.firstName} ${handleUser.lastName}`}</Typography>
+            <Tooltip title="Copy profile link" arrow>
+              <div>
+                <Typography
+                  variant="h5/medium"
+                  onClick={() => {
+                    copy(window.location.href);
+                    showToast('Profile link copied!');
+                  }}
+                >
+                  @{handleUser.handle}
+                </Typography>
+              </div>
+            </Tooltip>
+          </div>
+          <div className={styles.cardRank}>
+            <div className={styles.rank}>{getUserRank(handleUser.points)}</div>
+            <div className={styles.points}>
+              <LeaderboardIcon /> &nbsp;
+              {handleUser.points.toLocaleString()} All-Time Leaderboard Points
```

Comment:
"All-Time Leaderboard Points" feels kind of verbose. Maybe just "points" (since the points for this year aren't shown anywhere on the profile, I think) or "total points"?